go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/Azure/azure-container-networking
 
-go 1.22.0
-
-toolchain go1.22.7
+go 1.23
```

Comment:
Opinions on setting the toolchain to 1.23.1?

Reply:
now that the toolchain can upgrade itself based on the `toolchain` directive, how do we get something like dependabot to keep that line up to date?

Reply:
updated to 1.23.2